api/application/src/review/create.rs

```diff
@@ -1,22 +1,20 @@
 use diesel::prelude::*;
-use domain::models::review::{NewReview, Review};
+use domain::models::review::{NewReview, Review, NewReviewWithId};
 use infrastructure::ServerState;
-use rocket::{
-    response::status::{Conflict, Created},
-    serde::json::Json,
-    State,
-};
-use shared::response_models::ResponseMessage;
+use rocket::{http::Status, response::status::Created, serde::json::Json, State};
+use shared::{response_models::ResponseMessage, token::JWT};
 
 pub fn create_review(
     review: Json<NewReview>,
     state: &State<ServerState>,
-) -> Result<Created<String>, Conflict<String>> {
+    token: JWT,
+) -> Result<Created<String>, (Status, Json<ResponseMessage>)> {
     use domain::schema::reviews;
 
     let pooled = &mut state.db_pool.get().unwrap();
 
     let review = review.into_inner();
+    let review = NewReviewWithId::new(token.claims.sub, review);
 
     let review_count: i64 = match pooled.transaction(|c| {
```

Comment:
Review on a non-existent unit code leads to panic (due to foreign key constraint). Bug not introduced in this PR but just so we are aware.